tests/e2e/hcp_machine_pool_test.go

```diff
@@ -26,54 +103,244 @@ var _ = Describe("Create Machine Pool", labels.Feature.Machinepool, func() {
+			Expect(mpDesc.DesiredReplicas).Should(Equal(desiredReplicas))
+			Expect(mpDesc.InstanceType).Should(Equal(instanceType))
+
+			if isNodePoolGlobalCheck {
+				By("Check current replicas reach the desired replicas after creating a machine pool")
+				Eventually(checkReplicas).
+					WithArguments(mpName, false).
+					WithTimeout(constants.NodePoolCheckTimeout).
+					WithPolling(constants.NodePoolCheckPoll).
+					Should(BeTrue())
+			}
+
+			By("Scale a machine pool with unchanged parameters")
+			scaleNodePool(mpName, "unchanged", desiredReplicas, true)
 
-			Expect(mpDescription.AdditionalSecurityGroupIDs).To(Equal(strings.Join(sgIDs, ", ")))
+			By("Scale up a machine pool replicas from 1 to 2")
+			upReplicas := 2
+			scaleNodePool(mpName, "up", upReplicas, true)
+
+			By("Scale down a machine pool replicas from 2 to 1")
+			downReplicas := 1
+			scaleNodePool(mpName, "down", downReplicas, true)
+
+			By("Scale down a machine pool replicas to 0")
+			zeroReplica := 0
+			scaleNodePool(mpName, "down", zeroReplica, true)
+		},
+		Entry("For amd64 cpu architecture [id:60278]", constants.M5XLarge, constants.AMD),
+		Entry("For arm64 cpu architecture [id:60278]", constants.M6gXLarge, constants.ARM),
+	)
 
-			By("Create another machinepool without security groups and describe it")
-			mpName = "mp-72195-nsg"
-			_, err = rosaClient.MachinePool.CreateMachinePool(clusterID, mpName,
-				"--replicas", "1",
+	DescribeTable("Scale up/down a machine pool with invalid replica", labels.Critical, labels.Runtime.Day2,
+		func(instanceType string, updatedReplicas string, expectedErrMsg string) {
+			By("Create machinepool with instance " + instanceType)
+			mpName := common.GenerateRandomName("mp-60278", 2)
+			desiredReplicas := 1
+			_, err := rosaClient.MachinePool.CreateMachinePool(clusterID, mpName,
+				"--instance-type", instanceType,
+				"--replicas", fmt.Sprintf("%v", desiredReplicas),
 				"-y",
 			)
 			Expect(err).ToNot(HaveOccurred())
 			defer rosaClient.MachinePool.DeleteMachinePool(clusterID, mpName)
-			By("Check the machinepool detail by describe")
-			mpDescription, err = rosaClient.MachinePool.DescribeAndReflectNodePool(clusterID, mpName)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(mpDescription.AdditionalSecurityGroupIDs).To(BeEmpty())
+
+			_, err = rosaClient.MachinePool.EditMachinePool(clusterID, mpName,
+				"--replicas", fmt.Sprintf("%v", updatedReplicas),
+				"-y",
+			)
+			Expect(err.Error()).Should(ContainSubstring(expectedErrMsg))
+		},
+
+		Entry("Scale replica too large [id:60278]", constants.M52XLarge, "1000", "exceeds the maximum allowed"),
+		Entry("Scale replica to -1 [id:60278]", constants.M52XLarge, "-1", "Replicas must be a non-negative number"),
+		Entry("Scale replica to a char [id:60278]", constants.M52XLarge, "a", "invalid syntax"),
+	)
+
+	Describe("Scale up/down a machine pool enabling autoscale", func() {
+		When("with the valid min replica and max replica", func() {
```

Comment:
May I know why we have `When` here? If not too much benefits we can get from it or not too much required in the code, let's make code simple to read.

Reply:
Because there are many conditions under the same `Describe("Scale up/down a machine pool enabling autoscale"`, so use `When` to distinguish different sub-scenes.
```
Describe("Scale up/down a machine pool enabling autoscale"...
    When("with the valid min replica and max replica"...
    When("with the invalid min replica and max replica"...

```